service/history/task/timer_active_task_executor_test.go

```diff
@@ -801,7 +801,7 @@ func (s *timerActiveTaskExecutorSuite) TestDecisionScheduleToStartTimeout_Normal
 		return req.UpdateWorkflowMutation.ExecutionInfo.DecisionAttempt == 1 &&
 			req.UpdateWorkflowMutation.ExecutionInfo.DecisionScheduleID == 4 &&
 			req.UpdateWorkflowMutation.ExecutionInfo.NextEventID == 4 && // transient decision
-			len(req.UpdateWorkflowMutation.TimerTasks) == 1 // another schedule to start timer
+			len(req.UpdateWorkflowMutation.TasksByCategory[persistence.HistoryTaskCategoryTimer]) == 1 // another schedule to start timer
```

Comment:
can `TasksByCategory` be nil?

Reply:
it can be `nil`, but in this test, it's not `nil`